webapp/src/components/App/Singlestudy/explore/Modelization/Areas/Storages/StorageConfig/AdditionalConstraints/ConstraintForm/OccurrencesFE/OccurrencesTableDialog/index.tsx

```diff
@@ -0,0 +1,227 @@
+/**
+ * Copyright (c) 2025, RTE (https://www.rte-france.com)
+ *
+ * See AUTHORS.txt
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This file is part of the Antares project.
+ */
+
+import CustomScrollbar from "@/components/common/CustomScrollbar";
+import DataGridForm, {
+  type DataGridFormApi,
+  type DataGridFormProps,
+} from "@/components/common/DataGridForm";
+import BasicDialog from "@/components/common/dialogs/BasicDialog";
+import ConfirmationDialog from "@/components/common/dialogs/ConfirmationDialog";
+import NumberFE from "@/components/common/fieldEditors/NumberFE";
+import NumberSelectionsFE from "@/components/common/fieldEditors/NumberSelectionsFE";
+import type { ControlPlus, SubmitHandlerPlus } from "@/components/common/Form/types";
+import useConfirm from "@/hooks/useConfirm";
+import { HOURS_IN } from "@/utils/date/constants";
+import { Box, Button, ButtonGroup, Divider } from "@mui/material";
+import * as R from "ramda";
+import { useMemo, useRef, useState } from "react";
+import { useWatch } from "react-hook-form";
+import { useTranslation } from "react-i18next";
+import { type ConstraintValues } from "../../utils";
+import { validateOccurrences } from "../utils";
+import ColumnsResize from "./ColumnsResize";
+import {
+  getSelectedHoursByOccurrence,
+  gridDataToOccurrences,
+  occurrencesToGridData,
+  resizeRow,
+  type OccurrencesData,
+} from "./utils";
+
+interface Props {
+  open: boolean;
+  onClose: VoidFunction;
+  control: ControlPlus<ConstraintValues>;
+  occurrences: ConstraintValues["occurrences"];
+  onEdit: (occurrences: ConstraintValues["occurrences"]) => void;
+}
+
+function OccurrencesTableDialog({ open, onClose, onEdit, control, occurrences }: Props) {
+  const { t } = useTranslation();
+  const [columnCount, setColumnCount] = useState(occurrences.length);
+  const [offset, setOffset] = useState(0);
+  const dataGridApiRef = useRef<DataGridFormApi<OccurrencesData>>(null);
+  const [selectedHours, setSelectedHours] = useState<number[]>(R.range(1, HOURS_IN.WEEK + 1));
```

Comment:
This should be an empty array, otherwise you could affect 1-168 without having a range of hours selected before.

This can lead to overwriting existing configuration unexpectedly 

![Image](https://github.com/user-attachments/assets/e662b388-3b34-46f7-9655-25eecf1a41d8)


Also `offset` should be disabled if their is no hours selected in the field, to avoid such accidental manipulation of the existing config (if you change offset only you can "activate", "invert" etc)


![Image](https://github.com/user-attachments/assets/21d8a538-d8a5-4cd7-8648-ba3ec2a6ab29)

Reply:
For `offset` we dont need user feedback it's bug like to be able to do that, i think it should be disabled if hours is not "dirty". For the rest ok you can wait QA/user feedback